### ISSUE TITLE
skeema pull deletes dir for ignored schema

### ISSUE THREAD

**Issue by mugli** [skeema/skeema#55]
I have a `mydb_test` schema for test/dev environments. The production database obviously doesn't have the schema, so I've added it to `ignore-schema` option for production environment in config. 
Like this:
```
[production]
ignore-schema='tmp|mydb_test'
```

But every time I run `skeema pull -p production`, skeema deletes the directory with this message:
`Deleted directory .../schemas/mydb_test -- schema mydb_test no longer exists`

Is this the intended behaviour? How can I prevent skeema from deleting it?

**Comment by mugli** [comment]
Also I haven't found a recommended workflow for test environments: https://github.com/skeema/skeema/blob/master/doc/examples.md#advanced-configuration

In traditional db-migration tools with up-down migrations, we'd use a different schema name (like `mydb_test`) to apply the same migrations. The test schema usually exists on the same local database server alongside the development schema (`mydb`). 

What's the skeema way of doing so? How do I keep `mydb` and `mydb_test` in sync?

**Comment by mugli** [comment]
Forgot to mention, I'm using `skeema version 1.1.2-dev (snapshot build from source)`

**Comment by evanelias** [comment]
Thank you for the report! I can confirm the first part is a bug in `skeema pull`; I should have a fix released soon.

(I'm also tweaking the issue subject here just to clarify the *directory* is being deleted, not your actual *database* 😉)

Regarding your second question, the way to handle this in Skeema is to have a single directory, no need for symlinks. You just need to manually edit the .skeema file there to map the schema names correctly:
```
[production]
schema=mydb

[test]
schema=mydb_test
```

One question though -- do you just have one *production* schema, or several? If you only have one, I would recommend "flattening" the dir structure, so that there's no host-level parent dir, and the host/port/socket is defined in the same .skeema file as the schema name. Skeema will default to this automatically if you start with `skeema init ... --schema mydb`. With that setup, `skeema pull` will also work properly without needing to use `ignore-schema`.

If you have multiple production schemas, the correct layout is a bit harder to achieve, but still doable. If so, let me know more about your db layout, and I'm happy to help figure out the right config to match.

**Comment by mugli** [comment]
There are multiple production schemas. 

Different people in the team have other temporary schemas in their local db. The production has some of them too that are not being used anymore. Those all are declared in `ignore-schema` config. 

This is my current layout:
```
.
├── README.md
└── schemas
    ├── .skeema
    ├── mydb         
    │   ├── .skeema
    │   ├── table_name.sql
    ├── another_db
    │   ├── .skeema
    │   ├── table_name.sql
    └── mydb_test      <----- This is/was for develop/test environments only
        ├── .skeema
        ├── table_name.sql

```

So , if I understand it correctly now, I actually don't need a separate `mydb_test` directory at all. That's awesome, because I guess that'll solve my primary issue too (there will be no "extra" directory there that can be deleted during pulling from production) 👍

**Comment by evanelias** [comment]
Just released [Skeema v1.1.3](https://github.com/skeema/skeema/releases/tag/v1.1.3) which includes the `ignore-schema` fix from 212720e. Closing this out, but please feel free to reopen if you encounter any related questions/issues.
